### PR TITLE
Add `TestExecution.get_comments` API

### PR DESCRIPTION
**Pull request by asankov**
```
 $ curl 'http://localhost:8000/json-rpc/' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:80.0) Gecko/20100101 Firefox/80.0' -H 'Accept: application/json, text/javascript, */*; q=0.01' -H 'Accept-Language: en-US,en;q=0.7,bg;q=0.3' --compressed -H 'Content-Type: application/json' -H 'X-Requested-With: XMLHttpRequest' -H 'Origin: http://localhost:8000' -H 'Connection: keep-alive' -H 'Referer: http://localhost:8000/runs/1/' -H 'Cookie: csrftoken=BbgrN3nhlYb2WBnAKoC2wLhYoX5jYI5uYUXbfE2dPqJsONerUKKujMzoP74lLjcT; token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJVc2VyIjp7ImlkIjoiMSIsInVzZXJuYW1lIjoiYXNhbmtvdiIsImVtYWlsIjoiYXNhbmtvdiJ9LCJleHAiOjE1OTk0MjU1NjB9.SIM+Afy8Z06/ili3TYnue+JMXFqlrD4Ik59B/gQ4kU4=; djdt=hide; sessionid=rg9xh9nzswhy6azrd9oa3gnxlgzvxcc7; messages="[[\"__json_message\"\0541\05440\054\"You have 8 unapplied migration(s). See <a href=\\\"https://kiwitcms.readthedocs.io/en/latest/installing_docker.html#initial-configuration-of-running-container\\\">documentation</a>\"]\054[\"__json_message\"\0541\05440\054\"You have 8 unapplied migration(s). See <a href=\\\"https://kiwitcms.readthedocs.io/en/latest/installing_docker.html#initial-configuration-of-running-container\\\">documentation</a>\"]]:1kKT88:gyapDeol2fBC4OuholPnuRsaniXwdqGUZ1UWvxuWeQo"' -H 'Pragma: no-cache' -H 'Cache-Control: no-cache' --data-raw '{"jsonrpc":"2.0","method":"TestExecution.get_comments","params":["16"],"id":"jsonrpc"}' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1703  100  1617  100    86  14567    774 --:--:-- --:--:-- --:--:-- 15342
{
  "id": "jsonrpc",
  "jsonrpc": "2.0",
  "result": [
    {
      "id": 23,
      "object_pk": "16",
      "user_id": 1,
      "user": 1,
      "user_name": "atodorov",
      "user_email": "atodorov@otb.bg",
      "user_url": "",
      "comment": "We&#x27;ve found good merchandise and are in the process of ordering T-shirt for the team.",
      "submit_date": "2019-01-07T14:05:19.990"
    },
    {
      "id": 29,
      "object_pk": "16",
      "user_id": 1,
      "user": 1,
      "user_name": "atodorov",
      "user_email": "atodorov@otb.bg",
      "user_url": "",
      "comment": "All branded items finally done and delivered to team members.",
      "submit_date": "2019-01-26T12:36:35.595"
    },
    {
      "id": 876,
      "object_pk": "16",
      "user_id": 15,
      "user": 15,
      "user_name": "anton_sankov",
      "user_email": "anton_sankov@mail.com",
      "user_url": "",
      "comment": "something",
      "submit_date": "2020-09-20T21:18:05.849"
    },
    {
      "id": 877,
      "object_pk": "16",
      "user_id": 15,
      "user": 15,
      "user_name": "anton_sankov",
      "user_email": "anton_sankov@mail.com",
      "user_url": "",
      "comment": "something",
      "submit_date": "2020-09-20T21:18:31.804"
    },
    {
      "id": 878,
      "object_pk": "16",
      "user_id": 15,
      "user": 15,
      "user_name": "anton_sankov",
      "user_email": "anton_sankov@mail.com",
      "user_url": "",
      "comment": "something",
      "submit_date": "2020-09-20T21:18:41.418"
    },
    {
      "id": 879,
      "object_pk": "16",
      "user_id": 15,
      "user": 15,
      "user_name": "anton_sankov",
      "user_email": "anton_sankov@mail.com",
      "user_url": "",
      "comment": "something",
      "submit_date": "2020-09-20T21:19:07.215"
    },
    {
      "id": 880,
      "object_pk": "16",
      "user_id": 15,
      "user": 15,
      "user_name": "anton_sankov",
      "user_email": "anton_sankov@mail.com",
      "user_url": "",
      "comment": "something",
      "submit_date": "2020-09-21T21:13:56.261"
    }
  ]
}
```

### CODE REVIEW

File: tcms/rpc/api/testexecution.py [hunk 1]
```diff
@@ -76,6 +76,24 @@ def remove_comment(execution_id, comment_id=None):
...
+        Get all comments for selected test execution.
+
+        :param execution_id: PK of a TestExecution object
+        :type execution_id: int
+        :raises PermissionDenied: if missing *django_comments.view_comment* permission
+    """
+    execution = TestExecution.objects.get(pk=execution_id)
+    execution_comments = comments.get_comments(execution)
+    serialier = Serializer(execution_comments)
+    return serialier.serialize_queryset()
```
Comment (atodorov) [inline tcms/rpc/api/testexecution.py:94]: Don't use `Serializer`, it's slated for removal. Try list(execution_comments.values()), see Bug.filter() API methods.

File: tcms/rpc/api/testexecution.py [hunk 2]
```diff
@@ -76,6 +76,24 @@ def remove_comment(execution_id, comment_id=None):
     to_be_deleted.delete()
 
 
+@permissions_required('django_comments.view_comment')
+@rpc_method(name='TestExecution.get_comments')
+def get_comments(execution_id):
```
Comment (atodorov) [inline tcms/rpc/api/testexecution.py:81]: Also needs tests.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/rpc/api/testexecution.py [hunk 3]
```diff
@@ -76,6 +76,33 @@ def remove_comment(execution_id, comment_id=None):
...
+    """
+    .. function:: TestExecution.get_comments(execution_id)
```
Comment (atodorov) [inline tcms/rpc/api/testexecution.py:83]: Missing the RPC prefix in documentation.

File: tcms/rpc/tests/test_testexecution.py [hunk 4]
```diff
@@ -36,6 +36,26 @@ def test_add_comment_with_pk_as_int(self):
...
+        self.execution = TestExecutionFactory()
+        for comment in self.comments:
+            self.rpc_client.TestExecution.add_comment(self.execution.pk, comment)
+
+    def test_get_comment(self):
+        comments = self.rpc_client.TestExecution.get_comments(self.execution.pk)
+
+        self.assertEqual(len(self.comments), len(comments))
+        for comment in comments:
+            self.assertTrue(comment['comment'] in self.comments)
```
Comment (atodorov) [inline tcms/rpc/tests/test_testexecution.py:56]: Also needs a scenario covering the ObjectNotFound case when the method argument doesn't match an existing TestExecution.

**Review by atodorov — CHANGES_REQUESTED** [review]

**Review by atodorov — CHANGES_REQUESTED** [review]
Test scenarios look good now but see flake8 & pylint issues for the newly added files.

File: tcms/rpc/api/testexecution.py [hunk 5]
```diff
@@ -76,6 +76,23 @@ def remove_comment(execution_id, comment_id=None):
     to_be_deleted.delete()
 
 
+@permissions_required('django_comments.view_comment')
+@rpc_method(name='TestExecution.get_comments')
+def get_comments(execution_id):
```
Comment (atodorov) [inline tcms/rpc/api/testexecution.py:81]: You still have pylint issues for this method.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/rpc/api/testexecution.py (hunk 5 above)
Comment (asankov) [reply tcms/rpc/api/testexecution.py:81]: should be ok now

**Review by atodorov — APPROVED** [review]

**Pull request merged**